netmiko/base_connection.py

```diff
@@ -36,7 +37,8 @@ def __init__(self, ip='', host='', username='', password='', secret='', port=Non
                  device_type='', verbose=False, global_delay_factor=1, use_keys=False,
                  key_file=None, allow_agent=False, ssh_strict=False, system_host_keys=False,
                  alt_host_keys=False, alt_key_file='', ssh_config_file=None, timeout=8,
-                 session_timeout=60, keepalive=0, default_enter=None, response_return=None):
+                 session_timeout=60, keepalive=0, default_enter=None, response_return=None,
+                 serial_settings={}):
```

Comment:
You should make serial_settings default to None and then do:

```
if serial_settings is None:
    serial_settings = {}
```

Otherwise you run into an issue with Python mutables (i.e. we would get some unwanted side-effects).